From 86564051db45b466e5f667111ce00b5eeedc8fb6 Mon Sep 17 00:00:00 2001
From: Jason Summers <jason1@pobox.com>
Date: Wed, 12 Apr 2017 10:54:31 -0400
Subject: [PATCH] Reduced the max memory use and image size in 64-bit builds

This is a stopgap measure that should prevent some problems with excessive
resource use, and possibly integer overflow.
Related: issue #14
---
 src/imagew-internals.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/imagew-internals.h b/src/imagew-internals.h
index 3031359..0e520ef 100644
--- a/src/imagew-internals.h
+++ b/src/imagew-internals.h
@@ -19,8 +19,8 @@
 typedef double iw_tmpsample;
 
 #ifdef IW_64BIT
-#define IW_DEFAULT_MAX_DIMENSION 1000000
-#define IW_DEFAULT_MAX_MALLOC 2000000000000
+#define IW_DEFAULT_MAX_DIMENSION 40000
+#define IW_DEFAULT_MAX_MALLOC 2000000000
 #else
 #define IW_DEFAULT_MAX_DIMENSION 40000 // Must be less than sqrt(2^31).
 #define IW_DEFAULT_MAX_MALLOC 2000000000
